src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -224,8 +225,8 @@ public QuerySearchResult executeScan(ShardSearchRequest request) throws Elastics
                 throw new ElasticsearchIllegalArgumentException("aggregations are not supported with search_type=scan");
             }
             assert context.searchType() == SearchType.SCAN;
-            context.searchType(SearchType.COUNT); // move to COUNT, and then, when scrolling, move to SCAN
-            assert context.searchType() == SearchType.COUNT;
+            context.searchType(SearchType.QUERY_THEN_FETCH); // move to QUERY_THEN_FETCH, and then, when scrolling, move to SCAN
+            assert context.searchType() == SearchType.QUERY_THEN_FETCH;
```

Comment:
should this also set size = 0 for this first request so that it matches the old count functionality?